public/scripts/experiments/estimation/estimation.js

```diff
@@ -0,0 +1,465 @@
+        let mod_size = (round_num % 2 === 1)?
+            sub_cond.max_size * estimation_exp.PIXEL_TO_CM  : sub_cond.min_size * estimation_exp.PIXEL_TO_CM;
+        let mod_y = estimation_exp.calculate_y_position(mod_size);
+
+        this.curr_trial_data.is_ref_smaller = (round_num % 2 === 1);
+
+        if (Math.random() < 0.5) {
+            this.plot_shape(sub_cond.base_shape, chart, ref_size , ref_y, left_x, true);
+            this.plot_shape(sub_cond.mod_shape, chart, mod_size, mod_y, right_x, false);
+            this.curr_trial_data.is_ref_left = true;
+        } else {
+            this.plot_shape(sub_cond.mod_shape, chart, mod_size, mod_y, left_x, false);
+            this.plot_shape(sub_cond.base_shape, chart, ref_size, ref_y, right_x, true);
+            this.curr_trial_data.is_ref_left = false;
+        }
+
+    }
+
+    calculate_y_position(radius) {
+        let estimation_exp = this;
+        // y_margin is the distance from
+        let y_margin = estimation_exp.MARGIN * estimation_exp.PIXEL_TO_CM;
+        let range = [y_margin + radius / 2, window.innerHeight - y_margin - radius / 2];
+        console.log("radius: " + radius);
+        console.log(JSON.stringify(range));
+        let y_pos = Math.random() * (range[1] - range[0]) + range[0];
+        return y_pos;
+    }
+
+    plot_trials(){
+        console.log("plot_trials with index = " + this.curr_condition_index +
+            "round number" + this.curr_round_num);
+        this.plot_trial(this.curr_conditions_constants[this.curr_condition_index], this.curr_round_num);
+    }
+
+
+    plot_shape(shape, chart, radius, y_pos, x_pos, is_ref) {
+        switch (shape) {
+            case "circle":
+                this.plot_circle(chart, radius, y_pos, x_pos, is_ref);
+                break;
+            case "triangle":
+                this.plot_triangle(chart, radius, y_pos, x_pos, is_ref);
+                break;
+            case "rectangle":
+                this.plot_rectangle(chart, radius, y_pos, x_pos, is_ref);
+                break;
+        }
+    }
+    /**
+     * D3 code for plotting a circle.
+     *
+     * @param  chart      {object}
+     * @param  diameter     {number}
+     * @param  y_pos      {number}
+     * @param  x_pos      {number}
+     */
+    plot_circle(chart, diameter, y_pos, x_pos, is_ref) {
+        let exp = this;
+        let radius = diameter / 2;
+        chart.append("circle")
+            .attr("cx", x_pos)
+            .attr("cy", y_pos)
+            .attr("r", diameter / 2)
+            .attr("id", "circle_shape")
+            .attr("is_ref", is_ref)
+            .style("fill", exp.curr_trial_data.fill_color);
+        if (is_ref === false) {
+            d3.select("body")
+                .on("keydown", function () {
+                    let event = d3.event;
+                    // console.log(event);
+                    if (event.key === "m" || event.key === "z") {
```

Comment:
Might need to move the logic for key detection up into jsPsych - Wongi is abstracting all the graphing code so it is independent of the experiment logic so eventually we'd have to refactor this out anyway. 

I think if you have it move to the next trial upon key press, then identify the previous key press and use that to adjust up/down the size, we can get away w/ moving this out of the D3 code. I did this w/ Stevens in the `updated_estimation_correlation` method (though there's no refreshing for estimation).